src/backend/distributed/commands/multi_copy.c

```diff
@@ -986,6 +1023,49 @@ OpenCopyTransactions(CopyStmt *copyStatement, ShardConnections *shardConnections
 
 
 /*
+ * CanUseBinaryCopyFormat iterates over columns of the relation given in rowOutputState
+ * and looks for a column whose type is array of user-defined type or composite type. 
+ * If it finds such column, that means we cannot use binary format for COPY, because
+ * binary format sends Oid of the types, which are generally not same in master and
+ * worker nodes for user-defined types.
+ */
+static bool
+CanUseBinaryCopyFormat(TupleDesc tupleDescription, CopyOutState rowOutputState)
+{
+	bool useBinaryCopyFormat = true;
+	int totalColumnCount = tupleDescription->natts;
+	int columnIndex = 0;
+
+	for (columnIndex = 0; columnIndex < totalColumnCount; columnIndex++)
```

Comment:
I think it doesn't block the PR at all, I would only care if there was more code after the loop, but so I know what our preferred code style is... @anarazel wouldn't it be better to declare columnIndex inside the loop itself, on line 1039?